dns_server.go

```diff
@@ -37,6 +42,21 @@ func (d *dnsRecords) Query(data string) string {
 		d.RUnlock()
 		return r
 	}
+
+	if d.dnsWildcardEnabled {
+		limitCounter := 0
+		parts := strings.Split(data, ".")
```

Comment:
I think you can just use strings.SplitN here instead of the limit counter.

https://pkg.go.dev/strings#SplitN

Reply:
Will refactor this and also fix the missing call to `d.RUnlock()`